pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/TableRebalanceManager.java

```diff
@@ -69,6 +69,32 @@ public TableRebalanceManager(PinotHelixResourceManager resourceManager, Controll
     _executorService = executorService;
   }
 
+  /**
+   * Do a table rebalance dry-run for the table with the given name and type.
+   *
+   * @param tableNameWithType name of the table to rebalance
+   * @param rebalanceConfig configuration for the rebalance operation
+   * @param rebalanceJobId ID of the rebalance job, which is used to track the progress of the rebalance operation
+   * @return result of the rebalance dry-run operation
+   * @throws TableNotFoundException if the table does not exist
+   */
+  public RebalanceResult rebalanceTableDryRun(String tableNameWithType, RebalanceConfig rebalanceConfig,
+      String rebalanceJobId)
+      throws TableNotFoundException {
+    Preconditions.checkArgument(rebalanceConfig.isDryRun() || rebalanceConfig.isPreChecks());
```

Comment:
nit: pre-checks can only be enabled if dry-run is enabled. perhaps it is enough to check for just dry run here for that reason.

Reply:
Yeah that's what I had initially, but this test started to fail because it expects a proper rebalance result response in such a scenario (and only checking for dry run here would lead to a `500` error due to the `IllegalArgumentException`) - https://github.com/apache/pinot/blob/5d7fab1cc3177ce21fa2da1e1287f17c461c7df3/pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/TableRebalanceIntegrationTest.java#L978-L988

It seemed simpler to check for both here and allow `TableRebalancer` to do its thing - https://github.com/apache/pinot/blob/5d7fab1cc3177ce21fa2da1e1287f17c461c7df3/pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/TableRebalancer.java#L242-L248

Reply:
got it, makes sense